### PR TITLE
Feature/check 209 sources metadata

### PR DESCRIPTION
**Pull request by melsawy**

### CODE REVIEW

**Review by amoedoamorim — CHANGES_REQUESTED** [review]
I'm not sure if the annotated_type and parent_type mean the same thing. I see we have both in the code but I would favor a single usage, unless they mean different things. While we don't clarify this, I think the `annotated={source}` change is an important one.

File: src/app/components/task/Task.js [hunk 1]
```diff
@@ -489,7 +499,8 @@ class Task extends Component {
   }
 
   render() {
-    const { task, media } = this.props;
+    const { task: teamTask, media } = this.props;
+    const task = { ...teamTask };
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/task/Task.js:503]: Yes I got this error `Cannot add property cannotAct, object is not extensible` when working direct to `this.props.task`

File: src/app/components/team/CreateTeamTask.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ class CreateTeamTask extends React.Component {
       json_project_ids: task.json_project_ids,
       json_schema: task.jsonschema,
       fieldset: this.props.fieldset,
+      associated_type: this.props.associatedType,
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/team/CreateTeamTask.js:38]: It was `annotated_type` but Caio suggest `associated_type` in ticket description and I agreed about this name as we used `annotated_type` & `annotated_id` when refer to object but here we just need to know the `associated_type` either `ProjectMedia` or `Source`

File: src/app/components/annotations/Annotation.js [hunk 3]
```diff
@@ -320,10 +320,13 @@ class Annotation extends Component {
       task.reject_suggestion = vid;
     }
 
+    const parentType = this.props.annotatedType.replace(/([a-z])([A-Z])/, '$1_$2').toLowerCase();
+
     Relay.Store.commitUpdate(
       new UpdateTaskMutation({
         operation: 'suggest',
         annotated: this.props.annotated.project_media,
+        parent_type: parentType,
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/annotations/Annotation.js:329]: for this one I suggest to keep `parent_type` to be consist which commit other mutations like comment, flags, etc

File: src/app/components/task/Task.js [hunk 4]
```diff
@@ -185,10 +185,14 @@ class Task extends Component {
...
     const fields = {};
     fields[`response_${task.type}`] = response;
+
+    const parentType = task.annotated_type.replace(/([a-z])([A-Z])/, '$1_$2').toLowerCase();
+
     Relay.Store.commitUpdate(
       new UpdateTaskMutation({
         operation: 'answer',
         annotated: media,
+        parent_type: parentType,
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/task/Task.js:195]: I suggest to keep `parent_type` same other annotations mutation and file a ticket for refactor this field in all related mutations 
what do you think?

File: src/app/components/source/SourceInfo.js [hunk 5]
```diff
@@ -446,8 +450,318 @@ function SourceInfo({ source, team, onChangeClick }) {
           </Collapse>
         </Card>
       </Box>
+      <Tasks tasks={sourceMetadata} media={source} fieldset="metadata" noscroll />
```
[the remark this answers is not in the document]
Comment (melsawy) [reply src/app/components/source/SourceInfo.js:453]: for sure but it required to change in multiple files and extra testing for summit different type of annotations currently I am running in stable version and I'd like to merge to `develop` ASAP so I suggest to file a ticket with this refactor and work on it after merging stable version what do you think @amoedoamorim ?

File: src/app/components/task/Task.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (amoedoamorim) [reply src/app/components/task/Task.js:503]: Alright. I see, thanks Sawy, that makes sense. We should not mutate Relay data directly. Let's leave it as is for now and perhaps we can come up with a better pattern, maybe passing cannotAct as a separate prop on it's own.

**Pull request merged**